lib/cloud_controller/deployment_updater/actions/scale_down_old_process.rb

```diff
@@ -0,0 +1,35 @@
+module VCAP::CloudController
+  module DeploymentUpdater
+    module Actions
+      class ScaleDownOldProcess
+        attr_reader :deployment, :process, :app, :instances_to_scale_down
+
+        def initialize(deployment, process, instances_to_scale_down)
+          @deployment = deployment
+          @app = deployment.app
+          @process = process
+          @instances_to_scale_down = instances_to_scale_down
+        end
+
+        def call
+          if process.instances <= deployment.max_in_flight && is_interim_process?(process)
```

Comment:
**thought** Does this need to be here? Will this not happen as part of the eventual finalize deployment?

Reply:
It would probably eliminate the need for this action entirely